Firestore/core/src/firebase/firestore/nanopb/byte_string.cc

```diff
@@ -85,19 +102,39 @@ ByteString::~ByteString() {
   std::free(bytes_);
 }
 
+/* static */ ByteString ByteString::Copy(const pb_bytes_array_t* bytes) {
+  if (bytes == nullptr) {
+    return ByteString{};
+  } else {
+    return ByteString{MakeBytesArray(bytes->bytes, bytes->size)};
+  }
+}
+
 /* static */ ByteString ByteString::Take(pb_bytes_array_t* bytes) {
   return ByteString{bytes};
 }
 
+const uint8_t* ByteString::data() const {
+  static const uint8_t kEmpty[] = "";
+  return bytes_ ? bytes_->bytes : kEmpty;
```

Comment:
Question: seems like `pb_bytes_array_t` is guaranteed to be `uint8_t*`?

Reply:
`pb_bytes_array_t` is a struct with`pb_size_t` and `pb_byte_t[]` members. The latter is what we use as the address of the contents of the buffer.